File: backend/dataset/coinChange/coinChange_16.py
class Solution:
    def coinChange(self, coins: List[int], amt: int) -> int:
        if not amt:
                return 0
            coins.sort()
            if not coins:
                return -1
            q={0}
            stp=1
            while q:
                nwq=set()
                for a in q:
                    for c in coins:
                        if a+c>amt:
                            continue
                        if a+c==amt:
                            return stp
                        nwq.add(c+a)
            
                stp+=1
                q=nwq
            return -1                

File: backend/CDataset/coinChange/coinChange_16.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        vector<int> dp(amount+1,-1); dp[0] = 0;
        for(int i = 1;i<=amount;i++)
        {
            int mi = 100000000;
            for(auto a:coins)
                if(i>=a && dp[i-a]!=-1)  mi = min(mi,dp[i-a]);
            if(mi != 100000000)  dp[i] = 1+mi;
        }
        return dp[amount];
    }
};
